3515/1/files/gems_main.rb

```diff
@@ -0,0 +1,37 @@
+class GemsMain
+  attr_reader :threads
+  attr_reader :gems_arr
+  attr_reader :gems_table_arr
+
+  def main_process
+    gems_arr.gems_arr.each do |gem_name|
+      threads << Thread.new(gem_name) do
+        gem_url = GemsUrlLoader.new(gem_name)
+        next unless gem_url.url.include?('http')
+
+        param_getter(gem_url.url, gem_name)
+      end
+    end
+  end
+
+  def start(file = 'gems.yml')
+    gems_vars(file)
+    main_process
+    threads.each(&:join)
+  end
+
+  def param_getter(gem_url, gem_name)
+    gem_params = GemParamGetter.new(gem_url, gem_name)
+    gems_table_arr << gem_params.params
+  end
+
+  def table_builder(len = gems_arr.length)
```

Comment:
in default params used nil, different brackets, strings and etc. BUT not class of another method